Mathlib/SetTheory/Game/PGame.lean

```diff
@@ -1877,8 +1877,20 @@ theorem insertRight_insertLeft {x x' x'' : PGame} :
   cases x; cases x'; cases x''
   dsimp [insertLeft, insertRight]
 
-/-! ### Special pre-games -/
+/-! ### Dicotic pre-games -/
+
+/-- A game is dicotic if both players can move from every nonempty subposition of G. -/
+def dicotic (x : PGame) : Prop :=
+  (∀ (l : LeftMoves x),
+    (x.moveLeft l = 0) ∨
+    ((LeftMoves (x.moveLeft l) ≠ PEmpty) ∧
+    (RightMoves (x.moveLeft l)) ≠ PEmpty)) ∧
+  (∀ (r : RightMoves x),
+    (x.moveRight r = 0) ∨
+    ((LeftMoves (x.moveRight r) ≠ PEmpty) ∧
+    (RightMoves (x.moveRight r)) ≠ PEmpty))
```

Comment:
This doesn't seem right; it's not even recursive, and you shouldn't use type equality since there are many types of the same cardinality. I think the following is correct:
```
/-- A game is dicotic if both players can move from every nonempty subposition of G. -/
def IsDicotic (x : PGame) : Prop :=
  (IsEmpty x.LeftMoves ∧ IsEmpty x.RightMoves) ∨
  (Nonempty x.LeftMoves ∧ Nonempty x.RightMoves ∧
    (∀ l : x.LeftMoves, IsDicotic (x.moveLeft l)) ∧
    ∀ r : x.RightMoves, IsDicotic (x.moveRight r))
termination_by x
```
and I think for a PR on dicotic games to be accepted it (or another PR depending on it) must prove [this](https://github.com/leanprover-community/mathlib3/pull/14652#issuecomment-1151832079). This helps verify the definition is as intended.

Reply:
Indeed! I noticed the lack of a recursive definition, but stopped on dicotic games to prove that the equivalence class of games is formed in context of a lack of reversible and dominated options. Thanks for the syntax for a recursive definition, though!

That goal to prove the class dicotic games were equivalent to the class of small games was my goal for this PR.